src/Core/Managed/Shared/Extensibility/Implementation/TaskTimer.cs

```diff
@@ -17,15 +17,22 @@ namespace Microsoft.ApplicationInsights.Extensibility.Implementation
     /// <summary>
     /// Runs a task after a certain delay and log any error.
     /// </summary>
-    public class TaskTimer : IDisposable
+    public class TaskTimer
     {
+        private class DelayedWork
+        {
+            public DelayedWork(Func<Task> runFunc) { this.RunFunc = runFunc; this.IsCancellationRequested = false; }
+            public Func<Task> RunFunc { get; private set; }
```

Comment:
Could you name the `DelayedWork.RunFunc` property consistently with the `Start` method parameter?